src/sage/repl/attach.py

```diff
@@ -385,13 +396,13 @@ def add_attached_file(filename):
         []
     """
     sage.repl.inputhook.install()
-    fpath = os.path.abspath(filename)
-    attached[fpath] = os.path.getmtime(fpath)
+    fpath = Path(filename).resolve()
+    attached[fpath] = fpath.stat().st_mtime
```

Comment:
Couldn't we remove `resolve()` from line 399 to fix `/var` vs `/private/var` issue? Do we need `resolve()`?

Reply:
The idea is to store the attached files in fully resolved form, as before. The issues we have come from the bad behaviour of "tmp_filename" that should return Path objects in resolved form instead of strings. This should be changed later, but not here.

Reply:
> The idea is to store the attached files in fully resolved form, as before. 

Does `os.path.abspath(filename)` give a fully resolved form, or is `filename` already a fully resolved form?

Reply:
```
sage: t= tmp_filename(ext='.py')
sage: t
'/var/folders/ym/nw_hc8550mgc0v4r43lvswqh0000gn/T/tmpsnhsu7u8/tmp_3lmn7es3.py'
```
It seems that `tmp_filename` does not give a fully resolved form, before...

Reply:
I am on mac.

Reply:
> The idea is to store the attached files in fully resolved form, as before.

Before this PR:
```
sage: fn
'/var/folders/td/fw1q9ljs311ggyph77rs53_40000gn/T/tmp5hz2j1pi/tmp_asix90ys.py'
sage: attached_files()
['/var/folders/td/fw1q9ljs311ggyph77rs53_40000gn/T/tmp5hz2j1pi/tmp_asix90ys.py']
```
So the attached_files are not stored in fully resolved form. Right?

Then what is the reason (or benefits) of storing the attached files in fully resolved form, with this PR?

Reply:
We should at least store an absolute path, and not a relative one, as the user may change the current directory at any time. It seems to me that it is safe and a good idea to use `Path.resolve` to have the correct path.

Another possibility would be, according to

https://docs.python.org/fr/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

to use instead `Path.absolute()`, which should behave more like the existing code. Would you prefer that ?

Reply:
I think we should do either

(1) use `Path.absolute()` and keep current `tmp_filename()`

or

(2) use `Path.resolve()` and fix `tmp_filename()` to use fully resolved filename

instead of "monkey-patching" doctests for disconcerting `Path` and `tmp_filename()`.

(1) seems easier.